x-pack/plugins/ml/server/routes/data_frame_analytics.ts

```diff
@@ -13,12 +15,49 @@ import {
   dataAnalyticsExplainSchema,
   analyticsIdSchema,
   stopsDataFrameAnalyticsJobQuerySchema,
+  analyticsIdIndexSchema,
 } from './schemas/data_analytics_schema';
+import { IndexPatternHandler } from '../models/data_frame_analytics/index_patterns';
+
+function getIndexPatternId(context: RequestHandlerContext, patternName: string) {
+  const iph = new IndexPatternHandler(context.core.savedObjects.client);
+  return iph.getIndexPatternId(patternName);
+}
+
+function deleteIndexPatternById(context: RequestHandlerContext, indexPatternId: string) {
+  const iph = new IndexPatternHandler(context.core.savedObjects.client);
+  return iph.deleteIndexPatternById(indexPatternId);
+}
 
 /**
  * Routes for the data frame analytics
  */
 export function dataFrameAnalyticsRoutes({ router, mlLicense }: RouteInitialization) {
+  async function userCanDeleteIndex(
+    context: RequestHandlerContext,
+    destinationIndex: string
+  ): Promise<boolean> {
+    if (!mlLicense.isSecurityEnabled()) {
+      return true;
+    }
+    let privilege;
+    try {
+      privilege = await context.ml!.mlClient.callAsCurrentUser('ml.privilegeCheck', {
+        body: {
+          index: [
+            {
+              names: [destinationIndex], // uses wildcard
+              privileges: ['delete_index'],
+            },
+          ],
+        },
+      });
+      return privilege.index[destinationIndex].delete_index === true;
+    } catch (err) {
+      throw err;
```

Comment:
this try/catch is redundant as throwing `err` is the same as not having the catch

Reply:
Removed in https://github.com/elastic/kibana/pull/66934/commits/b867ef2cbcad2f423d1bace93ae74ec8ca45d648